sql/moz-fx-data-marketing-prod/acoustic/dataset_metadata.yaml

```diff
@@ -0,0 +1,10 @@
+friendly_name: Acoustic Derived
```

Comment:
`dataset_metadata.yaml` files are noncanonical for datasets outside of shared-prod/mozdata so should probably not be checked in (this might be worth a CI check). This is because the mechanism by which the dataset is managed (e.g. GA, fivetran, cloudops-infra TF, manually by DE) is often not using this metadata to provision the dataset.

In this case the canonical grant is to https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-marketing/tf/modules/resources/bigquery.tf#L56 but the `dataset_base_acl` is not used in provisioning that dataset.

Reply:
so to make sure I follow:
- the whole dataset is set to `mozilla-confidential` by DSRE ignoring the `dataset_metadata.yaml` line where it is specified as `workgroup:braze`
- That means implicitly all tables in the dataset are `mozilla-confidential` and we only need to specify it in `metadata.yaml` if we want to have something else
- `metadata.yaml` access is not additive with the default but will replace the default?

Reply:
>     * the whole dataset is set to `mozilla-confidential` by DSRE ignoring the `dataset_metadata.yaml` line where it is specified as `workgroup:braze`

No. I think there may be some confusion because there are two (really three) datasets here: `moz-fx-data-marketing-prod.acoustic` and `[moz-fx-data-shared-prod/mozdata].acoustic`. https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/acoustic/dataset_metadata.yaml#L10 is clearly annotating the user-facing dataset in `moz-fx-data-shared-prod` as `workgroup:mozilla-confidential`.

The only usage of `workgroup:braze` on this PR was on`moz-fx-data-marketing-prod.acoustic`. That metadata is technically not used to provision the dataset as stated above, so it's not canonical information (those files are only canonical for datasets under the `sql/moz-fx-data-shared-prod` directory.

>     * That means implicitly all tables in the dataset are `mozilla-confidential` and we only need to specify it in `metadata.yaml` if we want to have something else
This statement is true even if it doesn't follow from the previous statement as discussed above.
> 
>     * `metadata.yaml` access is not additive with the default but will replace the default?

`metadata.yaml` `workgroup_access` access is additive to `dataset_metadata.yaml` `workgroup_access`.

Reply:
I guess it's worth explicitly stating: an alternative implementation here is to lock down `moz-fx-data-shared-prod.acoustic`  to `workgroup:braze` and explicitly annotate every `workgroup:mozilla-confidential` view in it as such, but that didn't match the structure of the PR. My assumption is that we want to leverage the non-authorized nature of the views into the `workgroup:braze` dataset to block access to those users no in `workgroup:braze` while still keeping everything else open to `workgroup:mozilla-confidential`.